net_gen/src/if_tun.rs

```diff
@@ -26,10 +26,12 @@ impl<T> __IncompleteArrayField<T> {
     }
     #[inline]
     pub unsafe fn as_slice(&self, len: usize) -> &[T] {
+        // SAFETY: We trust the parameters
         unsafe { ::std::slice::from_raw_parts(self.as_ptr(), len) }
     }
     #[inline]
     pub unsafe fn as_mut_slice(&mut self, len: usize) -> &mut [T] {
+        // SAFETY: We trust the parameters
         unsafe { ::std::slice::from_raw_parts_mut(self.as_mut_ptr(), len) }
```

Comment:
This code was auto-generated by bindgen, so manual modifications should be avoided if possible. Can you please try to regenerate with newer version of bindgen and perhelps enable rust edition requirement and see if that would solve the clippy issue?

Reply:
Hm, generating the bindings is not straight forward and unfortunately, there is no `./script/bindgen.sh` or so helper for that.


I've tried:

- checkout into Linux source tree, v6.16
- `make headers_install ARCH=x86_64`
- `bindgen include/linux/if_tun.h --no-layout-tests -- -I$(pwd)/include -I$(pwd)/include/uapi -I$(pwd)/arch/x86/include`

but this fails with `Unable to generate bindings: clang diagnosed error: /home/pschuster/dev/linux/include/linux/compiler.h:390:10: fatal error: 'asm/rwonce.h' file not found`. I'm using `rust-bindgen 0.71.1`. I'm happy for help! @rbradford @likebreath

Reply:
I remove them for now in #7295 to unblock this.